Custom mf for user settings (#1802)
SAP/luigi: docs/luigi-core-api.md

Luigi.readUserSettings();
```

> Returns **[promise](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise)** a promise when a custom `readUserSettings` function in the settings.userSettings section of the Luigi configuration is implemented. It resolves a stored user settings object. If the promise is rejected the user settings dialog will also closed if the error object has a `closeDialog` property, e.g `reject({ closeDialog: true, message: 'some error' })`. In addition a custom error message can be logged to the browser console.

Reviewer: ```suggestion
Returns **[promise](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise)** a promise when `readUserSettings` function is defined in settings.userSettings object. It resolves a stored user settings object. If the promise is rejected, the user settings dialog will be closed in case `closeDialog` property is defined in the error object, e.g `reject({ closeDialog: true, message: 'some error' })`. In addition a custom error message can be logged to the browser console.
```
